dbms/src/Interpreters/ExpressionActions.h

```diff
@@ -162,11 +163,14 @@ class ExpressionActions
     void addInput(const ColumnWithTypeAndName & column);
     void addInput(const NameAndTypePair & column);
 
-    void add(const ExpressionAction & action);
+    void add(const ExpressionAction & action, const tipb::Expr * expr = nullptr);
 
     /// Adds new column names to out_new_columns (formed as a result of the added action).
     void add(const ExpressionAction & action, Names & out_new_columns);
 
+    // Set meta data for the last ExpressionAction
+    void setMetaData(const tipb::Expr & expr);
```

Comment:
when will this function be called?

Reply:
> when will this function be called?

https://github.com/xzhangxian1008/tiflash/blob/5fd73302a69f009c47c43d30a776e5f4b7293e68/dbms/src/Functions/IFunction.h#L404-L405

This function will be called when the called tidb function needs meta data.

Reply:
> when will this function be called?

I will delete useless functions.